mobile/native/src/ln_dlc/mod.rs

```diff
@@ -162,6 +169,134 @@ pub fn run(data_dir: String) -> Result<()> {
+    node.inner.sync();
+
+    let wallet_balances = node
+        .get_wallet_balances()
+        .context("Failed to get wallet balances")?;
+
+    let WalletHistories {
+        on_chain,
+        off_chain,
+    } = node
+        .get_wallet_histories()
+        .context("Failed to get wallet histories")?;
+
+    let on_chain = on_chain.iter().map(|details| {
+        let net_sats = details.received as i64 - details.sent as i64;
+
+        let (flow, amount_sats) = if net_sats >= 0 {
+            (api::PaymentFlow::Outbound, net_sats as u64)
+        } else {
+            (api::PaymentFlow::Inbound, net_sats.unsigned_abs())
+        };
+
+        let (status, timestamp) = match details.confirmation_time {
+            Some(BlockTime { timestamp, .. }) => (api::Status::Confirmed, timestamp),
+
+            None => {
+                (
+                    api::Status::Pending,
+                    // Unconfirmed transactions should appear towards the top of the history
+                    OffsetDateTime::now_utc().unix_timestamp() as u64,
+                )
+            }
+        };
+
+        let wallet_type = api::WalletType::OnChain {
+            txid: details.txid.to_string(),
+        };
+
+        api::WalletHistoryItem {
+            flow,
+            amount_sats,
+            timestamp,
+            status,
+            wallet_type,
+        }
+    });
+
+    let off_chain = off_chain.iter().filter_map(|details| {
+        let amount_sats = match details.amount_msat {
+            Some(msat) => msat / 1_000,
+            // Skip payments that don't yet have an amount associated
+            None => return None,
+        };
+
+        let status = match details.status {
+            ln_dlc_node::node::HTLCStatus::Pending => api::Status::Pending,
+            ln_dlc_node::node::HTLCStatus::Succeeded => api::Status::Confirmed,
+            // TODO: Handle failed payments
+            ln_dlc_node::node::HTLCStatus::Failed => return None,
+        };
+
+        let flow = match details.flow {
+            ln_dlc_node::node::PaymentFlow::Inbound => api::PaymentFlow::Inbound,
+            ln_dlc_node::node::PaymentFlow::Outbound => api::PaymentFlow::Outbound,
+        };
+
+        let timestamp = details.timestamp.unix_timestamp() as u64;
+
+        let wallet_type = api::WalletType::Lightning {
+            payment_hash: hex::encode(details.payment_hash.0),
+        };
+
+        Some(api::WalletHistoryItem {
+            flow,
+            amount_sats,
+            timestamp,
+            status,
+            wallet_type,
+        })
+    });
+
+    let trades = {
+        let orders = crate::db::get_filled_orders()
+            .context("Failed to get filled orders; skipping update")?;
+
+        orders.into_iter().enumerate().map(|(i, order)| {
+            let flow = if i % 2 == 0 {
+                api::PaymentFlow::Outbound
+            } else {
+                api::PaymentFlow::Inbound
+            };
```

Comment:
🔧 I should have probably added a TODO about how insanely hacky this is. Although it works very well for the time being since we only support an order history that goes `[open, close, open, close, open, ...]`.

Reply:
😄 I guess we should capture these hacks in tickets that are attached to feature epics eventually.

The feature epic here is: Support extend/reduce the position - once we do that this hack has to be fixed.